sdk/trace/tracetest/example_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package tracetest_test
+
+import (
+	"context"
+	"fmt"
+
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+)
+
+func Example_simulateWorkflow() {
+	ctx := context.Background()
+
+	// Set up an in-memory span recorder and tracer provider.
+	recorder := tracetest.NewSpanRecorder()
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithSpanProcessor(recorder),
+	)
+
+	defer func() {
+		_ = tp.Shutdown(ctx)
+	}()
+
+	tracer := tp.Tracer("example/workflow")
+
+	// Parent span "workflow"
+	ctx, workflowSpan := tracer.Start(ctx, "workflow")
+
+	// Child span "step-1"
+	_, stepSpan := tracer.Start(ctx, "step-1")
```

Comment:
Why are we creating two spans? I think it over-complicates the example for people who just want to learn how to use NewSpanRecorder.